pkg/providers/instancetype/suite_test.go

```diff
@@ -767,10 +766,70 @@ var _ = Describe("InstanceType Provider", func() {
 			})
 		})
 	})
+	Context("Provider List MaxPods", func() {
+		BeforeEach(func() {
+			ctx = options.ToContext(ctx, test.Options())
+		})
+		It("should set pods equal to MaxPods in the AKSNodeClass when specified", func() {
+			maxPods := int32(150)
+			nodeClass.Spec.MaxPods = lo.ToPtr(maxPods)
+
+			instanceTypes, err := azureEnv.InstanceTypesProvider.List(ctx, nodeClass)
+			Expect(err).NotTo(HaveOccurred())
+			ExpectCapacityPodsToMatchMaxPods(instanceTypes, maxPods)
```

Comment:
Thinking about it some more, can you add another NodeClass with a different MaxPods in the same test? I suspect it will fail, the synthetic key for instance type cache does not include MaxPods, would likely get incorrect cache hit

https://github.com/Azure/karpenter-provider-azure/blob/53f7670982be737ea00e7a5d308f0e696c3c0214/pkg/providers/instancetype/instancetypes.go#L114-L119

Reply:
This is a great catch!! Updated the code now.